conans/cli/command.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+
+from conans.cli.cli import SmartFormatter
+from conans.errors import ConanException
+
+
+def info_handler(func):
+    def decorator(*args, **kwargs):
+        ret = {"error": None, "data": None}
+        try:
+            ret["data"] = func(*args, **kwargs)
+        except Exception as exc:
+            ret["error"] = exc
+            raise
+        finally:
+            return ret
+
+    return decorator
+
+
+class ConanCommand(object):
+    def __init__(self, method, group=None, **kwargs):
+        self._formatters = {}
+        self._allowed_formatters = ["cli", "json"]
+        for kind, action in kwargs.items():
+            if kind not in self._allowed_formatters:
+                raise ConanException("Formatter '{}' not allowed. Allowed formatters: {}"
+                                     .format(kind, " ".join(self._allowed_formatters)))
+            if callable(action):
+                self._formatters[kind] = action
+        self._group = group or "Misc commands"
+        self._name = method.__name__.replace("_", "-")
+        self._method = method
+        if method.__doc__:
+            self._doc = method.__doc__
+        else:
+            raise ConanException("No documentation string defined for command: '{}'. Conan "
+                                 "commands should provide a documentation string explaining "
+                                 "its use briefly.".format(self._name))
+        self._parser = argparse.ArgumentParser(description=self._doc,
+                                               prog="conan {}".format(self._name),
+                                               formatter_class=SmartFormatter)
```

Comment:
Here we can add the _formatters_:

```suggestion
                                               formatter_class=SmartFormatter)
        self._parser.add_argument('-o', '--output', default="cli", action=OnceArgument,
                                  help="Select the output format: cli, json.")
```

List only those available.

Maybe it is easier to add a `parser` outside that consume some arguments (the command itself, the `output`, `-v/--version`, `--help`,...) and pass remaining arguments to the parser instantiated by each command.